### PR TITLE
goferish refactor and fixes

### PR DESCRIPTION
**Pull request by bartossh**
Changes and fixes.

- Introduce struct to be instantiated,
- Add methods on pointer receiver,
- Fix the number of slices allocations to the bare minimum,
- Naming according to go standards,
- Remove naming redundancy (neurax.NeuraxXXX -> neurax.XXX),
- Introduce default config,
- allow for context passing and safe server shutdown,
- allow for safe goroutine closing (some of them, more goroutines needs a context for safe exit),
- removed pythonic ideas (clunked the code unnecessarily), pythonic ideas work well for small prototype projects, not for reusable, well-performing, testable, and safe code.

I would like to refactor more but not in one go.

Please if there is anything that is somehow unclear I would be happy to clarify this. 
I think this library may have great potential but needs some architecture changes. 
This isn't easy when building things from scratch.

### CODE REVIEW

File: neurax.go [hunk 1]
```diff
@@ -24,61 +28,15 @@ import (
...
-}*/
+const (
+	tearDownServer = time.Millisecond * 500
+	listenerTick   = time.Millisecond * 10
```
Comment (bartossh) [inline neurax.go:33]: It may be smaller, 1 ms even

**Comment by wintrmvte** [comment]
Hi, first of all, thank you for your contribution <3 I will merge your PR and will also mirror changes in naming conventions to README as soon as I have some spare minute. Everything is clear and well-written, yet I have some hard time understanding the `listenerTick` variable. Is it used for graceful timeout handling?

**Comment by bartossh** [comment]
listenerTick will allow reading from  `conn, err := l.Accept() ` that is run in an infinite loop. But if  `context.Context ` will be canceled, it will then stop the loop immediately. Otherway you are keeping goroutine in an infinite loop and without the possibility of stoping this loop. So this is going to run forever. (leaking resources)

File: neurax.go (hunk 1 above)
Comment (p506) [reply neurax.go:33]: Ok

**Pull request merged**